tests/filesystem_tests/test_hdfs_handler.py

```diff
@@ -53,6 +54,40 @@ def test_klist_not_exist(self):
 
         os.environ['PATH'] = path
 
+    def create_dummy_keytab(self, dummy_username):
+        dummy_password = "123"
+        keytab_path = "user.keytab"
+        command = "(echo 'addent -password -p {}@{} -k 1 -e rc4-hmac' &&\
+                  sleep 1 && echo {} && echo write_kt {}) \
+                  | ktutil".format(dummy_username, "dummy_realm",
+                                   dummy_password, keytab_path)
+        pipe = subprocess.Popen(command, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE, shell=True)
+        out, err = pipe.communicate()
+        return keytab_path
+
+    def test_get_principel_name_from_keytab(self):
```

Comment:
This method name indicates it being intended for unittest of `def _parse_klist_keytab_output(output):` somehow. Also practical unittest on parsing the output of `klist` had better be separated from integrated test of `create_handler()`.